dask_expr/_collection.py

```diff
@@ -828,6 +829,13 @@ def set_index(
         elif other == self.index.name:
             return self
 
+        if not sort:
+            if npartitions is not None:
+                raise ValueError(
+                    "Specifying npartitions with sort=False is not supported. Call `repartition` afterwards."
```

Comment:
May also make sense to raise an error when `sorted` is `True` (below).

Reply:
lets rather move it after sorted, makes a bit more sense, since performance is the same, but specifying sorted enables setting the divisions

Reply:
Yeah, you can order things however you feel is best. I'm only noting that it is problematic (perhaps wrong) for the user to specify `npartitions` when `sort=False` **or** when `sorted=True` (since we ignore `npartitions` for both of these cases).

Reply:
Ah sorry, misunderstood you earlier. Updated this now.